src/Workspaces/Core/MSBuild/MSBuild/DiagnosticReporter.cs

```diff
@@ -58,7 +58,7 @@ public void Report(DiagnosticLog log)
         {
             foreach (var logItem in log)
             {
-                Report(DiagnosticReportingMode.Log, GetMSBuildFailedMessage(logItem.ProjectFilePath, logItem.ToString()));
+                Report(new WorkspaceDiagnostic(logItem.Kind, GetMSBuildFailedMessage(logItem.ProjectFilePath, logItem.ToString())));
             }
```

Comment:
Does it also make sense to update line 68 to return the raw msbuild message directly? MsBuild error messages usually already contain the path and location and it feels a bit repetitive and to wrap it. E.g., the message from test case:

`Msbuild failed when processing the file 'C:\Users\yufeih\AppData\Local\Temp\RoslynTests\6c6362f9-f471-491b-9271-f25c1949e25a\CSharpProject\CSharpProject.csproj' with message: C:\Users\yufeih\AppData\Local\Temp\RoslynTests\6c6362f9-f471-491b-9271-f25c1949e25a\CSharpProject\CSharpProject.csproj: (49, 5): This is a test of the msbuild error system. This is only a test.`

Reply:
Would the paths be different if the error came from a .targets file?

Reply:
Yes it would.

> [Failure] Msbuild failed when processing the file '/Users/yufeih/docfx/samples/csharp/src/CSharp.csproj' with message: /Users/yufeih/docfx/samples/csharp/src/a.targets: (4, 5): This is a test of the msbuild error system. This is a warning.